tds/src/main/java/thredds/server/wfs/WFSController.java

```diff
@@ -49,14 +55,137 @@ private void getCapabilities(PrintWriter out, HttpServletRequest hsreq) {
 	@RequestMapping("**")
 	public void httpHandler(HttpServletRequest hsreq, HttpServletResponse hsres) {
 		try {
+			
 			PrintWriter wr = hsres.getWriter();
-			String request = hsreq.getParameter("REQUEST");
+			List<String> paramNames = new LinkedList<String>();
+			Enumeration<String> paramNamesE = hsreq.getParameterNames();
+			while(paramNamesE.hasMoreElements()) paramNames.add(paramNamesE.nextElement());
+			
+			// Prepare parameters
+			String request = null;
+			String version = null;
+			String service = null;
+			
+			/* Look for parameter names to assign values
+			 * in order to avoid casing issues with parameter names (such as a mismatch between reQUEST and request and REQUEST).
+			 */
+			
+			for(String paramName : paramNames) {
+				if(paramName.equalsIgnoreCase("REQUEST")) {
+					request = hsreq.getParameter(paramName);
+				}
+				
+				if(paramName.equalsIgnoreCase("VERSION")) {
+					version = hsreq.getParameter(paramName);
+				}
+				
+				
+				if(paramName.equalsIgnoreCase("SERVICE")) {
+					service = hsreq.getParameter(paramName);
+				}
+			}
+			
+			// The SERVICE parameter is required. If not specified, is an error (throw exception through XML).
+			if(service != null) {
+
+				// For the WFS servlet it must be WFS if not, write out an InvalidParameterValue exception.
+				if(!service.equalsIgnoreCase("WFS")) {
+					WFSExceptionWriter owsExcept = new WFSExceptionWriter("WFS Server error. SERVICE parameter must be of value WFS.", service, "InvalidParameterValue", hsres);
+					owsExcept.write();
+					return;
+				}
+				
+			}
+			
+			else {
+				WFSExceptionWriter owsExcept = new WFSExceptionWriter("WFS server error. SERVICE parameter is required.", "request", "MissingParameterValue", hsres);
+				owsExcept.write();
+				return;
+			}
 			
 			// The REQUEST Parameter is required. If not specified, is an error (throw exception through XML).
 			if(request != null) {
-				if(request.equals(WFSRequestType.GetCapabilities.toString())) {
+				
+				// Only go through version checks if NOT a Get Capabilities request, the VERSION parameter is required for all operations EXCEPT GetCapabilities section 7.6.25 of WFS 2.0 Interface Standard
+				if(!request.equalsIgnoreCase(WFSRequestType.GetCapabilities.toString())) {
+				
+					if(version != null ) {
+						// If the version is not failed report exception VersionNegotiationFailed, from OGC Web Services Common Standard section 7.4.1
+						
+						List<String> versionParts = new ArrayList<String>();
+						String currentPart = "";
+						
+						//Interestingly, string split has problems with "." so split it manually

```

Comment:
Interesting! I did try escaping the period previously with no effect. But perhaps that's because I did not escape escape it (i.e. "\." instead of "\\."). We could go with split I suppose although the current implementation is working as is.

Reply:
This will be updated in a commit in the near future.